string  action
string  dataName
int32   dataVal
---
int32   data
